### DISCUSSION TITLE
Citekeys in different collections

### DISCUSSION THREAD

**Discussion by professionalAnxious5** [retorquere/zotero-better-bibtex#2509] [Q&A] [answered]
Hello, 
sorry if this is a very bacis q. I am new to this. 
Why do pinned citekeys not sync across different collections? 
I create unique citkeys for every entry in my library and use these entries in different collections, shared and not shared. But the Citekey is not syncing (after refreshing) which I do not understand, because I thought this is the idea of a pinned citekey. Maybe I did not understand this right. My goal is to set a fixed citekey once for one entry.
Thank you in advance

**Comment by retorquere** [comment]
> Why do pinned citekeys not sync across different collections?

If they are pinned, they will sync (I take it you mean libraries/groups? sync isn't really terminology that applies to collections).

> I create unique citkeys for every entry in my library and use these entries in different collections, shared and not shared. But the Citekey is not syncing (after refreshing) which I do not understand, because I thought this is the idea of a pinned citekey.

But are they in fact pinned? Do they appear in the `extra` field? If not, they are not pinned.

**Reply by professionalAnxious5** [reply]
Thank you! 
Yes, they are pinned, they appear in the extra field. Collections are just sub-categories of libraries, sorry for the terminology, I use Zotero in German. 
Does is matter where I change the citekey? But actually it does not matter, when I change the citekey in my main library it does not sync in the group library and vice versa.

**Reply by retorquere** [reply]
Do you mean that

1. they are present in the `extra` field on one side, they do sync to the other side to the `extra` field, but they are not used as citekeys on the other side, or
2. they are present in the `extra` field on one side, but they do not sync to the other side to the `extra` field (and are for that reason not used as the citekey)

Because 2. would be a problem with Zotero sync, not with BBT. 1. would be a BBT problem, and then you need to open an issue instead of a discussion.

It does not matter where you make the change. What is in the `extra` field must always be the active citekey, regardless of how it got there.

**Comment by professionalAnxious5** [comment]
second. 
Thank you for nailing this down. I will get in touch with Zotero.

**Reply by retorquere** [reply]
do they sync if you uninstall BBT?

**Reply by professionalAnxious5** [reply]
If I do that won't I be losing all my customised citekeys?

**Reply by retorquere** [reply]
No, BBT never destroys the database. It only removes the behavior, not the data.

**Reply by retorquere** [reply]
Any new items you create while BBT is not installed will get a key as soon as you start up with BBT installed. Pinned keys survive even if you delete the BBT database (**not** the Zotero database), because what is in the `extra` field is always used as the citekey.

**Reply by professionalAnxious5** [reply]
does not sync either after I uninstalled BBT. did re-install it, did not fix in either.

**Reply by retorquere** [reply]
Thanks for testing that. That makes it unambiguously a Zotero sync problem.

**Reply by professionalAnxious5** [reply, marked as answer]
ok, Zotero support says, that items in different libraries are totally separated and can never sync/refresh. Then I got this wrong, becaus I thought the goal of a pinned citekey was to define it interlibrary and not to set it for each new library again.

**Reply by retorquere** [reply]
Oh you mean between *different* libraries/groups. Yeah that's not how Zotero works. There isn't even a concept of "same item" between libraries.
